tutorials/3d/procedural_geometry/introduction.rst

```diff
@@ -0,0 +1,106 @@
+rotation, position etc.). 
+
+If you are going to draw the same object many times, it can be helpful to use a MultiMesh with 
+a MultiMeshInstance. The MultiMeshInstance is used to draw meshes thousands of times very 
+cheaply. It takes advantage of hardware rendering in order to do so. The drawback with 
+using a MultiMeshInstance is that you are limited to one material for all instances. It uses an
+instance array to store different colors and transformations for each instance, but they all the
+instances use the same material.
+
+What a Mesh is
+--------------
+
+A Mesh is composed of one or more surfaces. A surface is an array composed of multiple sub-arrays
+containing vertices, normals, UVs etc. Normally the process of constructing surfaces and meshes is
+hidden from the user in the :ref:`VisualServer <class_VisualServer>`, but with ArrayMeshes, the user can construct a Mesh
+manually by passing in an array containing the surface information.
+
+Surfaces
+^^^^^^^^
+
+Each surface has its own material, alternatively, you can override the material for all surfaces
+in the Mesh when you use a MeshInstance using ``MeshInstance.override_material``.
+
+Surface array
+^^^^^^^^^^^^^
+
+The surface array has a is an array of length ``ArrayMesh.ARRAY_MAX``. Each position in the array is
+filled with a sub-array containing per-vertex information. For example the array located at 
+``ArrayMesh.ARRAY_NORMAL`` is a :ref:`PoolVector3Array <class_PoolVector3Array>` of vertex normals. 
+
+The surface array can be indexed or non-indexed. Creating a non-indexed array is as easy as not assigning
+an array at the index ``ArrayMesh.ARRAY_INDEX``. A non-indexed array stores unique vertex information for
+every triangle, meaning that when two triangle share a vertex, the vertex is duplicated in the array. An 
+indexed surface array only stores vertex information for each unique vertex and then also stores an array 
+of indices which maps out how to construct the triangles from the vertex array. In general, using an indexed 
+array is faster, but it means you have to share vertex data between triangles, which is not always desired 
+(e.g. when you want per-face normals).
+
+Tools
+-----
+
+Godot provides different ways of accessing and working with geometry. More information on each will
+be provided in the following tutorials.
+
+ArrayMesh
+^^^^^^^^^
+
+The ArrayMesh resource extends the Mesh to add a few different quality of life functions, and most 
+importantly, the ability to construct a Mesh surface through scripting.
+
+For more information about the ArrayMesh, please see the `ArrayMesh tutorial` (coming soon).
+
+MeshDataTool
+^^^^^^^^^^^^
+
+The MeshDataTool is a resource that converts Mesh data into arrays of vertices, faces, and edges that can 
+be modified at runtime.
+
+For more information about the MeshDataTool, please see the `MeshDataTool tutorial` (coming soon).
```

Comment:
This line should be removed. Just add the reference once the tutorial is actually there :)